scripts/gather-licenses.js

```diff
@@ -52,6 +52,23 @@ async function gatherLicenses(startPath) {
                     if (!path.isAbsolute(resolved)) {
                         continue; // Node.js builtin
                     }
+                    // Immediately try to find root of the dependency with the
+                    // package.json to avoid finding extraneous package.json
+                    // files inside the distribution (happens sometimes in
+                    // weird cases, see stylis library for an example)
+                    resolved =
+                      (await findUp(
+                        async (directory) => {
+                          const pkgJson = path.join(directory, 'package.json');
+                          if (
+                            directory.endsWith(dep) &&
+                            (await findUp.exists(pkgJson))
+                          ) {
+                            return pkgJson;
+                          }
+                        },
+                        { cwd: resolved }
+                      )) || resolved;
```

Comment:
I think this logically belongs more to the place where `pkgJson` is looked up? i.e. I’d probably replace the `pkgUp` call there with one to `findUp` that verifies that the package.json has the needed metadata, if I’m understanding the problem correctly?

Reply:
Sure, makes sense, will change

Reply:
Changed in 7222f63, how does it look?